packages/netlify-cms-core/src/reducers/deploys.ts

```diff
@@ -0,0 +1,51 @@
+import { produce } from 'immer';
+import {
+  DEPLOY_PREVIEW_REQUEST,
+  DEPLOY_PREVIEW_SUCCESS,
+  DEPLOY_PREVIEW_FAILURE,
+  DeploysAction,
+} from '../actions/deploys';
+
+export type Deploys = {
+  [key: string]: {
+    isFetching: boolean;
+    url?: string;
+    status?: string;
+  };
+};
+
+const defaultState: Deploys = {};
+
+const deploys = produce((state: Deploys, action: DeploysAction) => {
+  switch (action.type) {
+    case DEPLOY_PREVIEW_REQUEST: {
+      const { collection, slug } = action.payload;
+      state[`${collection}.${slug}`] = {
+        ...state[`${collection}.${slug}`],
```

Comment:
This is required so when clicking the `Check for preview" button, it doesn't disappear